tools/make-v8.sh

```diff
@@ -18,22 +18,36 @@ if [ "$ARCH" = "s390x" ] || [ "$ARCH" = "ppc64le" ]; then
   # set paths manually for now to use locally installed gn
   export BUILD_TOOLS=/home/iojs/build-tools
   export LD_LIBRARY_PATH="$BUILD_TOOLS:$LD_LIBRARY_PATH"
-  # Avoid linking to ccache symbolic links as ccache decides which
-  # binary to run based on the name of the link (we always name them gcc/g++).
-  # shellcheck disable=SC2154
-  CC_PATH=`command -v "$CC" gcc | grep -v ccache | head -n 1`
-  # shellcheck disable=SC2154
-  CXX_PATH=`command -v "$CXX" g++ | grep -v ccache | head -n 1`
   rm -f "$BUILD_TOOLS/g++"
   rm -f "$BUILD_TOOLS/gcc"
-  ln -s "$CXX_PATH" "$BUILD_TOOLS/g++"
-  ln -s "$CC_PATH" "$BUILD_TOOLS/gcc"
+  # V8's build config looks for binaries called `gcc` and `g++` if not using
+  # clang. Ensure that `gcc` and `g++` point to the compilers we want to
+  # invoke, creating symbolic links placed at the front of PATH, if needed.
+  # Avoid linking to ccache symbolic links as ccache decides which binary
+  # to run based on the name of the link (i.e. `gcc`/`g++` in our case).
+  # shellcheck disable=SC2154
+  if [ "$CC" != "" ] && [ "$CC" != "gcc" ]; then
+    # shellcheck disable=SC2154
+    CC_PATH=`command -v "$CC" gcc | grep -v ccache | head -n 1`
+    ln -s "$CC_PATH" "$BUILD_TOOLS/gcc"
+  fi
+  # shellcheck disable=SC2154
```

Comment:
I don't believe this comment is necessary.

```suggestion
```

Reply:
It came out of https://github.com/nodejs/node/runs/5413643748?check_suite_focus=true
```
Error: /home/runner/work/node/node/tools/make-v8.sh:33:9: CXX is referenced but not assigned.
```
https://github.com/nodejs/node/blob/b8c645b1bfce6f53995e00b7c7e4a153dbd26632/tools/make-v8.sh#L33

Reply:
With this line removed:
```console
$ tools/lint-sh.js tools/ --from-npx
Running Shell scripts checker...

In /home/rlau/sandbox/github/node/tools/make-v8.sh line 33:
  if [ "$CXX" != "" ] && [ "$CXX" != "g++" ]; then
        ^--^ SC2154 (warning): CXX is referenced but not assigned.

For more information:
  https://www.shellcheck.net/wiki/SC2154 -- CXX is referenced but not assigned.
```

Reply:
I've updated to incorporate the other two suggestions but kept this one because the linter fails otherwise.